### PR TITLE
[JAVA-961] Dynamic configuration is not valid at some time

### PR DESCRIPTION
**Pull request by laijianbin**
Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Make sure there is a [JIRA issue](https://https://servicecomb.atlassian.net/projects/JAV/issues) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [ ] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Format the pull request title like `[JAV-XXX] Fixes bug in ApproximateQuantiles`, where you replace `JAV-XXX` with the appropriate JIRA issue.
 - [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [ ] Run `mvn clean install` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---
When delete  the dynamic configuration  through Config Center, then ServiceComb will use  RoundRobin as the default ,after a while, sending the same configuration i hava deleted ,the configuration will not valid

### CODE REVIEW

File: handlers/handler-loadbalance/src/main/java/io/servicecomb/loadbalance/LoadbalanceHandler.java [hunk 1]
```diff
@@ -350,4 +350,16 @@ private void loadFilter(String filter, LoadBalancer lb) {
       }
     }
   }
+
+  private boolean isNotChanged(String p, String s) {
```
Comment (wujimin) [inline handlers/handler-loadbalance/src/main/java/io/servicecomb/loadbalance/LoadbalanceHandler.java]: use meaning name

File: handlers/handler-loadbalance/src/main/java/io/servicecomb/loadbalance/LoadbalanceHandler.java [hunk 2]
```diff
@@ -350,4 +350,16 @@ private void loadFilter(String filter, LoadBalancer lb) {
...
+    if (p == null && s == null) {
+      return this.policy == null && this.strategy == null;
+    } else if (p == null) {
```
Comment (wujimin) [inline handlers/handler-loadbalance/src/main/java/io/servicecomb/loadbalance/LoadbalanceHandler.java]: already return, no need else

File: handlers/handler-loadbalance/src/main/java/io/servicecomb/loadbalance/LoadbalanceHandler.java (hunk 1 above)
Comment (wujimin) [inline handlers/handler-loadbalance/src/main/java/io/servicecomb/loadbalance/LoadbalanceHandler.java]: maybe "isChanged" is better

Comment (wujimin) [inline handlers/handler-loadbalance/src/main/java/io/servicecomb/loadbalance/LoadbalanceHandler.java]: UT

File: handlers/handler-loadbalance/src/main/java/io/servicecomb/loadbalance/LoadbalanceHandler.java [hunk 3]
```diff
@@ -350,4 +350,16 @@ private void loadFilter(String filter, LoadBalancer lb) {
...
+      return this.policy == null && s.equals(this.strategy);
+    } else if (s == null) {
+      return this.strategy == null && p.equals(policy);
+    } else {
+      return p.equals(policy) && s.contentEquals(strategy);
```
Comment (wujimin) [inline handlers/handler-loadbalance/src/main/java/io/servicecomb/loadbalance/LoadbalanceHandler.java]: why strategy need special contentEquals?

File: handlers/handler-loadbalance/src/main/java/io/servicecomb/loadbalance/LoadbalanceHandler.java (hunk 1 above)
Comment (wujimin) [inline handlers/handler-loadbalance/src/main/java/io/servicecomb/loadbalance/LoadbalanceHandler.java]: so complex?
it seems old logic is simpler.

**Comment by WillemJiang** [comment]
@laijianbin The JIRA number is not write, could you double check it?

File: handlers/handler-loadbalance/src/main/java/io/servicecomb/loadbalance/LoadbalanceHandler.java (hunk 3 above)
Comment (eric-lee-ltk) [reply handlers/handler-loadbalance/src/main/java/io/servicecomb/loadbalance/LoadbalanceHandler.java]: maybe just implement a compare method like the below:
```
private static boolean isEqual(String str1, String str2) {
    return (str1 == null ? str2 == null : str1.equals(str2));
}
```
Then we can simply call `isEqual(policy, this.policy) && isEqual(strategy, this.strategy)` to get the result without so many if/else.

Comment (laijianbin) [reply handlers/handler-loadbalance/src/main/java/io/servicecomb/loadbalance/LoadbalanceHandler.java]: @eric-lee-ltk  This method（isNotChanged()） only use for a judgement . i think there will not much change.so i think no need.

Comment (laijianbin) [reply handlers/handler-loadbalance/src/main/java/io/servicecomb/loadbalance/LoadbalanceHandler.java]: @wujimin  oh! i make a mistake

File: handlers/handler-loadbalance/src/main/java/io/servicecomb/loadbalance/LoadbalanceHandler.java (hunk 1 above)
Comment (laijianbin) [reply handlers/handler-loadbalance/src/main/java/io/servicecomb/loadbalance/LoadbalanceHandler.java]: Yeah, the old logic is simpler. But after delete  configuration ,ServiceComb uses RoundRobin Rule as default ,then i  push configuration like RandomRule .ServiceComb still uses the  default configuration.

Comment (laijianbin) [reply handlers/handler-loadbalance/src/main/java/io/servicecomb/loadbalance/LoadbalanceHandler.java]: Okay, i will try

File: handlers/handler-loadbalance/src/main/java/io/servicecomb/loadbalance/LoadbalanceHandler.java (hunk 3 above)
Comment (eric-lee-ltk) [reply handlers/handler-loadbalance/src/main/java/io/servicecomb/loadbalance/LoadbalanceHandler.java]: @laijianbin What you intend to do in the `isNotChanged` looks like just compare whether two variables are equal or not. You implementation looks too complicated!

File: handlers/handler-loadbalance/src/main/java/io/servicecomb/loadbalance/LoadbalanceHandler.java (hunk 1 above)
Comment (wujimin) [reply handlers/handler-loadbalance/src/main/java/io/servicecomb/loadbalance/LoadbalanceHandler.java]: yes, old logic is simpler but had some problems
but new logic is too confusion

i think eric-lee-ltk's suggestion is good.

Comment (laijianbin) [reply handlers/handler-loadbalance/src/main/java/io/servicecomb/loadbalance/LoadbalanceHandler.java]: oh, i think use "isChanged" will more complex

**Comment by laijianbin** [comment]
I pull a new request ,so i close this PR

**Pull request closed without merging**